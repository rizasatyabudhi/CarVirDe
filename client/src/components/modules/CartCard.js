import React, { Component } from "react";

export class CarCart extends Component {
  render() {
    return (
      <div class="col s12">
        <div class="card-panel hoverable hover">
          <h5>Car Order</h5>
          <div className="row">
            <div className="col s10">
              <div className="card-content  ">
                <div className="row">
                  <div className="col s6">
                    <ul>
                      <li>
                        <div className="row">
                          <div className="col s12">
                            <span className="category">Tipe Mobil &nbsp;:</span>
                            {this.props.tipe}
                          </div>
                        </div>
                      </li>
                      <li>
                        <div className="row">
                          <div className="col s12">
                            <span className="category">Merk Mobil &nbsp;:</span>
                            {this.props.merk}
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                  <div className="col s6">
                    <ul>
                      <li>
                        <div className="row">
                          <div className="col s12">
                            <span className="category">
                              Tahun &nbsp;&nbsp;&nbsp;&nbsp;:
                            </span>
                            {this.props.tahun}
                          </div>
                        </div>
                      </li>
                      <li>
                        <div className="row">
                          <div className="col s12">
                            <span className="category">Transmisi &nbsp;: </span>
                            {this.props.transmisi}
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div className="col s2">
              <h5>Rp 500.000</h5>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export class AttractionCart extends Component {
  render() {
    return (
      <div class="col s12">
        <div class="card-panel hoverable">
          <h5>Attraction Order</h5>
          <div className="row">
            <div className="col s10">
              Lorem ipsum dolor sit, amet consectetur adipisicing elit.
              Laudantium delectus est, aliquam minima a excepturi, facere sequi
              similique doloremque eos molestias quod ad eius temporibus saepe
              totam nihil asperiores iste!
            </div>
            <div className="col s2">
              <h5>Rp 200.000</h5>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export class TotalPriceCart extends Component {
  render() {
    return (
      <div>
        <div class="row">
          <div class="col s12 center-align">
            <div
              class="card-panel teal z-depth-4"
              style={{ width: "300px", display: "inline-block" }}
            >
              <span class="white-text">
                <h5>
                  Total Price: <br /> Rp 700.000
                </h5>
              </span>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
