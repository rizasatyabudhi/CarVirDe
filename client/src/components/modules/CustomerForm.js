import React, { Component } from "react";

export default class CustomerForm extends Component {
  render() {
    return (
      <div class="row container">
        <form class="col s6">
          <div class="row">
            <div class="input-field col s12">
              <input type="password" placeholder="password" />
            </div>
          </div>
          <div class="row">
            <div class="input-field col s12">
              <input type="email" placeholder="email" />
            </div>
          </div>
        </form>
      </div>
    );
  }
}
