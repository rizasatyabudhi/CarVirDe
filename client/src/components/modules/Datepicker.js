import React, { Component } from "react";
import DatePicker from "react-datepicker";
import moment from "moment";

export class datepickerTimesheet extends Component {
  static defaultProps() {
    placeholder: "";
  }

  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(date) {
    this.props.input.onChange(moment(date).format("YYYY-MM-DD"));
    // this.props.input.onChange(moment(date).format(`DD-${MMM.toUpperCase()}-YY`))
  }

  render() {
    const { input, placeholder, meta: { touched, error } } = this.props;

    return (
      <div style={this.props.style}>
        {this.props.inputName ? (
          <h2 className="input-name">{this.props.inputName}</h2>
        ) : null}
        {this.props.inputDesc ? (
          <h2 className="input-desc">{this.props.inputDesc}</h2>
        ) : null}
        <DatePicker
          {...input}
          placeholder={placeholder}
          dateFormat="YYYY-MM-DD"
          // selected={input.value ? moment(input.value, `DD-MMM${.toUpperCase()}-YY`) : null}
          selected={input.value ? moment(input.value, "YYYY-MM-DD") : null}
          onChange={this.handleChange}
          minDate={moment().subtract(2, "month")}
          maxDate={moment()}
        >
          {this.props.children}
        </DatePicker>
      </div>
    );
  }
}
