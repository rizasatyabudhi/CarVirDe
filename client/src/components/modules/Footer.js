import React, { Component } from "react";

export default class Footer extends Component {
  render() {
    return (
      <div>
        <footer class="page-footer deep-purple darken-3 lighten-1">
          <div class="container">
            <div class="row">
              <div class="col l6 s12">
                <h5 class="white-text">CarVirde</h5>
                <p class="grey-text text-lighten-4">Your Traveling Solution</p>
              </div>
              <div class="col l4 offset-l2 s12">
                <i className="material-icons">contact_phone</i>
                <h5
                  style={{
                    display: "inline-block",
                    marginLeft: "20px",
                    fontSize: "15px"
                  }}
                  class="white-text"
                >
                  021-7574829
                </h5>
                <ul>
                  {/*  <li>
                    <a class="grey-text text-lighten-3" href="#!">
                      Link 1
                    </a>
                  </li>
                  <li>
                    <a class="grey-text text-lighten-3" href="#!">
                      Link 2
                    </a>
                  </li>
                  <li>
                    <a class="grey-text text-lighten-3" href="#!">
                      Link 3
                    </a>
                  </li>
                  <li>
                    <a class="grey-text text-lighten-3" href="#!">
                      Link 4
                    </a>
                  </li>
                  */}
                </ul>
              </div>
              <div class="col l4 offset-l2 s12">
                <i className="material-icons">email</i>
                <h5
                  style={{
                    display: "inline-block",
                    marginLeft: "20px",
                    fontSize: "15px"
                  }}
                  class="white-text"
                >
                  contact-support@carvirde.com
                </h5>
                <ul />
              </div>
            </div>
          </div>
          <div class="footer-copyright">
            <div class="container">
              Â© 2017 CarVirde
              <a
                class="grey-text text-lighten-4 right"
                href="https://github.com/rizasatyabudhi/CarVirDe"
              >
                https://github.com/rizasatyabudhi/CarVirDe
              </a>
            </div>
          </div>
        </footer>
      </div>
    );
  }
}
