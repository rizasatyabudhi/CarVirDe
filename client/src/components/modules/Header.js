import React, { Component } from "react";
import { Link } from "react-router-dom";

export default class Header extends Component {
  render() {
    return (
      <nav style={{ marginBottom: "20px" }}>
        <div class="nav-wrapper deep-purple darken-3 lighten-1">
          <Link to={"/"} className="brand-logo" style={{ marginLeft: "1em" }}>
            CarVirde
          </Link>
          <ul class="right">
            <li>
              <Link to={"register"}>Register</Link>
            </li>
            <li>
              <Link to={"login"}>Login</Link>
            </li>
            <li>
              <Link to={"rentCar"}>Rent a Car</Link>
            </li>
            <li>
              <Link to={"addCar"}>Add Car</Link>
            </li>
            <li>
              <Link to={"attraction"}>Attraction</Link>
            </li>
            <li>
              <Link to={"addAttraction"}>Add Attraction</Link>
            </li>
            <li>
              <Link to={"cart"}>
                <i className="material-icons">shopping_cart</i>
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}
