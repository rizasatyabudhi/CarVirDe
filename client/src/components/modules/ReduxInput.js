import React, { Component } from "react";

export default class ReduxInput extends Component {
  render() {
    return (
      <div>
        <input type="text" {...this.props.input} />
      </div>
    );
  }
}
