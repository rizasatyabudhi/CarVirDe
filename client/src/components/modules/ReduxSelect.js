import React, { Component } from "react";
export class ReduxSelect extends Component {
  constructor() {
    super();
    this.state = {
      error: null
    };
  }
  render() {
    return (
      <select
        {...this.props.select}
        {...this.props.custom}
        {...this.props.input}
        placeholder={this.props.placeholder}
      >
        {this.props.children}
      </select>
    );
  }
}
