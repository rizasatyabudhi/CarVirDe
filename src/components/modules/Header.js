import React, { Component } from "react";
import { Link } from "react-router-dom";
import { store } from "../../index";
import { connect } from "react-redux";

class Header extends Component {
  componentDidUpdate() {}
  render() {
    const { orders = {}, auth = {} } = this.props;
    return (
      <nav style={{ marginBottom: "20px" }}>
        <div class="nav-wrapper deep-purple darken-3 lighten-1">
          <Link
            to={"/"}
            className="brand-logo logo-font"
            style={{ marginLeft: "1em" }}
          >
            CarVirde
          </Link>
          <ul class="right">
            <li>
              <Link to={"register"}>Register</Link>
            </li>
            <li>
              <Link to={"login"}>Login</Link>
            </li>
            {store.getState().auth === "rent" && (
              <li>
                <Link to={"addCar"}>Tambah Mobil Rental</Link>
              </li>
            )}
            {store.getState().auth === "attraction" && (
              <li>
                <Link to={"addAttraction"}>Tambah Tempat Wisata</Link>
              </li>
            )}
            {store.getState().auth === "customer" && (
              <li>
                <Link to={"cart"}>
                  <i className="material-icons">shopping_cart</i>
                </Link>
              </li>
            )}
          </ul>
        </div>
      </nav>
    );
  }
}

function mapStateToProps(state) {
  return { state };
}
export default connect(mapStateToProps, {})(Header);
