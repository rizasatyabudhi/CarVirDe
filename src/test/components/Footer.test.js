import React from "react";
import { shallow } from "enzyme";
import Footer from "../../components/modules/Footer";
import { CarCart } from "../../components/modules/CartCard";

test("should render Footer", () => {
  const wrapper = shallow(<Footer />);
  expect(wrapper).toMatchSnapshot();
});
