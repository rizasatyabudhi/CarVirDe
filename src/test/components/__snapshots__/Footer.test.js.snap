// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer 1`] = `
<div>
  <footer
    class="page-footer deep-purple darken-3 lighten-1"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col l6 s12"
        >
          <h5
            class="white-text logo-font"
          >
            CarVirde
          </h5>
          <p
            class="grey-text text-lighten-4"
          >
            Your Traveling Solution
          </p>
        </div>
        <div
          class="col l4 offset-l2 s12"
        >
          <i
            className="material-icons"
          >
            contact_phone
          </i>
          <h5
            class="white-text"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "15px",
                "marginLeft": "20px",
              }
            }
          >
            021-7574829
          </h5>
          <ul />
        </div>
        <div
          class="col l4 offset-l2 s12"
        >
          <i
            className="material-icons"
          >
            email
          </i>
          <h5
            class="white-text"
            style={
              Object {
                "display": "inline-block",
                "fontSize": "15px",
                "marginLeft": "20px",
              }
            }
          >
            contact-support@carvirde.com
          </h5>
          <ul />
        </div>
      </div>
    </div>
    <div
      class="footer-copyright"
    >
      <div
        class="container"
      >
        Â© 2017 CarVirde
        <a
          class="grey-text text-lighten-4 right"
          href="https://github.com/rizasatyabudhi/CarVirDe"
        >
          www.carvirde.com
        </a>
      </div>
    </div>
  </footer>
</div>
`;
